TODO/src/components/Home.js

```diff
@@ -0,0 +1,173 @@
+import styles from '../styles';
+import React, {useState} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import {
+  addTodo,
+  deleteTodo,
+  completeTodo,
+  editTodo,
+  activeUser,
+  completed,
+  all,
+  DeleteAll,
+} from '../redux/actions/action';
+import {
+  Text,
+  View,
+  KeyboardAvoidingView,
+  Platform,
+  TextInput,
+  TouchableOpacity,
+  Modal,
+  FlatList,
+  SafeAreaView,
+} from 'react-native';
+import Task from './task';
+
+const Home = () => {
+  const dispatch = useDispatch();
+
+  const state = useSelector(state => state);
+
+  const [Modalopen, setModalopen] = useState(false);
+  const [selecteditem, setSelectedItem] = useState([]);
+
+  const [task, setTask] = useState('');
+
+  const [newValue, setNewValue] = useState([]);
+
+  const handleAddTask = () => {
+    dispatch(addTodo(task));
+    setTask('');
+  };
+
+  const update = i => {
+    setModalopen(true);
+    setNewValue({text: i.text, id: i.id});
+  };
+
+  const edit = e => {
+    setModalopen(false);
+    dispatch(editTodo(newValue));
+  };
+  const edithandler = event => {
+    setNewValue(prev => ({...prev, text: event.nativeEvent.text}));
+  };
+
+  console.log(state);
+  const renderItem = ({item}) => {
+    return (
+      <View>
+        <Task
+          id={item.id}
+          completed={item.completed}
+          text={item.text}
+          completeTask={() => dispatch(completeTodo(item.id))}
+          deleteTask={() => dispatch(deleteTodo(item.id))}
+          update={() => {
+            update(item);
+          }}
+        />
+      </View>
```

Comment:
Pass minimum number of props

Reply:
<Task
id={item.id}
task={item}
update={update}
/>

Reply:
<Task
          id={item.id}
          task={item}
          update={() => {update(item)}
        />